receiver/apachesparkreceiver/metadata.yaml

```diff
@@ -4,4 +4,674 @@ status:
+    description: The result of the executor tasks for which the metric was recorded.
+    type: string
+    enum:
+      - completed
+      - failed
+  job_task_result:
+    description: The result of the job tasks for which the metric was recorded.
+    type: string
+    enum:
+      - completed
+      - failed
+      - skipped
+  job_stage_result:
+    description: The result of the job stages for which the metric was recorded.
+    type: string
+    enum:
+      - completed
+      - failed
+      - skipped
+  direction:
+    description: Whether the metric is in regards to input or output operations.
+    type: string
+    enum:
+      - in
+      - out
+  source:
+    description: The source from which data was fetched for the metric.
+    type: string
+    enum:
+      - local
+      - remote
+  location:
+    description: The location of the memory for which the metric was recorded..
+    type: string
+    enum:
+      - on_heap
+      - off_heap
+  state:
+    description: The state of the memory for which the metric was recorded.
+    type: string
+    enum:
+      - used
+      - free
+  scheduler_waiting:
+    description: Whether the stages for which the metric was recorded are waiting to be run by the DAGScheduler.
+    type: bool
+  scheduler_running:
+    description: Whether the stages for which the metric was recorded are currently being run by the DAGScheduler.
+    type: bool
+  pool_memory_type:
+    description: The type of pool memory for which the metric was recorded.
+    type: string
+    enum:
+      - direct
+      - mapped
+  gc_type:
+    description: The type of the garbage collection performed for the metric.
+    type: string
+    enum:
+      - major
+      - minor
+
+metrics:
+  #stage
+  spark.stage.task.active:
+    description: Number of active tasks in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: false
+      value_type: int
+    unit: "{ task }"
+    attributes:
+      [stage_active, stage_complete, stage_pending, stage_failed]
+  spark.stage.task.results:
+    description: Number of tasks with a specific result in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: "{ task }"
+    attributes:
+      [stage_active, stage_complete, stage_pending, stage_failed, stage_task_result]
+  spark.stage.executor.run_time:
+    description: Amount of time spent by the executor in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: ms
+    attributes:
+      [stage_active, stage_complete, stage_pending, stage_failed]
+  spark.stage.executor.cpu_time:
+    description: CPU time spent by the executor in this stage.
+    enabled: true
+    sum:
+      aggregation: cumulative
+      monotonic: true
+      value_type: int
+    unit: ns
+    attributes:
+      [stage_active, stage_complete, stage_pending, stage_failed]
```

Comment:
I think these attributes should only be attached to a single metric that describes the state of the stage. We don't need to report it for every stage metric.